cpp/memilio/config.h

```diff
@@ -27,6 +27,7 @@
 
 #include "memilio/config_internal.h"
 #include <type_traits>
+#include "ad/ad.hpp"
 
 using ScalarType = double;
 
```

Comment:
Since we don't use any properties of the ad type we should get away with only a forward declaration. This way we avoid including the ad header in basically all files (because config.h is), which may reduce compilation time or binary size.

```suggestion

using ScalarType = double;

namespace ad
{
namespace internal
{
// Forward declaration used for support of ad types in Limits.
template <class AD_TAPE_REAL, class DATA_HANDLER>
struct active_type;
} // namespace internal
} // namespace ad
```

Reply:
That is a very good point, thank you!